### DISCUSSION TITLE
Does Pipeline.from_pretrained downloads the model each time or uses a cache?

### DISCUSSION THREAD

**Discussion by lsshawn** [pyannote/pyannote-audio#1267] [Q&A]
When we run `Pipeline.from_pretrained` using hugging face, does it download the model every time? Or will it read from my cache?

I'm trying to make this code run faster:

```
pipeline = Pipeline.from_pretrained(
        "pyannote/segmentation",
        use_auth_token="TOKEN",
    )

for wav in wavFiles:
    pipeline(wav)
```

**Comment by hbredin** [comment]
Please read the [FAQ](https://github.com/pyannote/pyannote-audio/blob/develop/FAQ.md#can-i-use-gated-models-and-pipelines-offline) explaining how to use models and pipelines in an offline manner.
